frontend/components/vault/WithdrawDaiFromVat.vue

```diff
@@ -0,0 +1,116 @@
+<template>
+    <BasePanel :current-state="currentStateAndTitle.name">
+        <template #title>{{ currentStateAndTitle.title }}</template>
+        <TextBlock v-if="isExplanationsShown">
+            After liquidating a vault, the DAI-based liquidation incentive will end up in your VAT account. One more
+            transaction is required to move that DAI to your wallet.
+        </TextBlock>
+        <div class="my-4">
+            <WalletAuthorizationCheckPanel
+                :disabled="!isWalletConnected"
+                :wallet-address="walletAddress"
+                :is-wallet-authorized="isWalletAuthorized"
+                :is-explanations-shown="isExplanationsShown"
+                :is-loading="isAuthorizing"
+                @authorizeWallet="$emit('authorizeWallet')"
+            />
+        </div>
+        <div class="flex justify-end mt-2 gap-5">
+            <BaseButton
+                :disabled="!isWalletConnected || !isWalletAuthorized || !hasDaiToWithdraw || isWithdrawing"
+                @click="$emit('manageVat')"
+            >
+                Withdraw DAI partially
+            </BaseButton>
+            <BaseButton
+                type="primary"
+                :disabled="!isWalletConnected || !isWalletAuthorized || !hasDaiToWithdraw"
+                :is-loading="isWithdrawing"
+                @click="$emit('withdrawAllDaiFromVat')"
+            >
+                <span v-if="hasDaiToWithdraw">
+                    Withdraw <FormatCurrency :value="daiVatBalance" currency="DAI" /> from VAT
+                </span>
+                <span v-else> Nothing to withdraw yet </span>
+            </BaseButton>
+        </div>
+    </BasePanel>
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import BigNumber from 'bignumber.js';
+import { VaultTransactionState } from 'auctions-core/src/types';
+import BaseButton from '~/components/common/inputs/BaseButton.vue';
+import TextBlock from '~/components/common/other/TextBlock.vue';
+import FormatCurrency from '~/components/common/formatters/FormatCurrency.vue';
+import BasePanel from '~/components/common/other/BasePanel.vue';
+import WalletAuthorizationCheckPanel from '~/components/panels/WalletAuthorizationCheckPanel.vue';
+
+export default Vue.extend({
+    name: 'WithdrawDAIPanel',
+    components: {
+        BaseButton,
+        BasePanel,
+        TextBlock,
+        FormatCurrency,
+        WalletAuthorizationCheckPanel,
+    },
+    props: {
+        walletAddress: {
+            type: String,
+            default: undefined,
+        },
+        daiVatBalance: {
```

Comment:
This is something Valia and I discussed a while ago. This component does not work without `daiVatBalance` and a few other props. Therefore we should stop making these props optional and just add a `v-if` on the component in the parent to ensure it is hidden when in this case no wallet is connected. 

This is better for testing, as if it is ever not properly hidden or a value is not correctly passed, we get a error thrown during the tests. 

(If the prop is only optional we may never know we forgot to pass it somewhere)